### PR TITLE
Spend flow test

### PR DESCRIPTION
**Pull request by doodyparizada**
add a test for spend flow

### CODE REVIEW

File: scripts/src/internal/app.ts [hunk 1]
```diff
@@ -30,3 +34,11 @@ createRoutes(app);
 app.use(notFoundHandler);
 // catch errors
 app.use(generalErrorHandler);
+
+// initializing db and models
+initModels().then(msg => {
+	logger.debug("init db", { msg });
```
Comment (nitzantomer) [inline scripts/src/internal/app.ts]: why `, { msg }` and not just `, msg`?

File: scripts/src/internal/app.ts [hunk 2]
```diff
@@ -30,3 +34,11 @@ createRoutes(app);
...
+	initPaymentCallbacks(logger).then(res => {
+		logger.debug("init payment result", { res });
```
Comment (nitzantomer) [inline scripts/src/internal/app.ts]: same here with the `, { res }`

File: scripts/src/internal/services.ts [hunk 3]
```diff
@@ -69,3 +71,15 @@ export async function paymentFailed(payment: CompletedPayment, reason: string, l
...
 	logger.info(`failed order with payment <${payment.id}, ${payment.transaction_id}>`);
 }
+
+/**
+ * register to get callbacks for incoming payments for all the active offers
+ */
+export async function initPaymentCallbacks(logger: LoggerInstance): Promise<Watcher> {
+	const offers = await Offer.find({ type: "spend" }); // get all active spend offers
+	// create a list of unique addresses
+	const addresses: string[] = removeDups(offers.map(offer => offer.blockchainData.recipient_address));
```
Comment (nitzantomer) [inline scripts/src/internal/services.ts]: no need for the type annotation here, can be: 
```const addresses = removeDups(offers.map(offer => offer.blockchainData.recipient_address));```

File: scripts/src/mock_client.ts [hunk 4]
```diff
@@ -31,13 +32,19 @@ const STELLAR = new Stellar("testnet");
 class Client {
 
 	public authToken: AuthToken;
-	private keyPair: StellarSdk.KeyPair;
+	private keyPair: StellarSdk.Keypair = null;
```
Comment (nitzantomer) [inline scripts/src/mock_client.ts]: why is it important to intialize it to `null`?

File: scripts/src/mock_client.ts [hunk 5]
```diff
@@ -46,25 +53,40 @@ class Client {
...
 
-	public async pay(recipientAddress: string, amount: number): Promise<StellarSdk.TransactionResult> {
+	public async pay(recipientAddress: string, amount: number, orderId: string): Promise<TransactionRecord> {
 		const op = StellarSdk.Operation.payment({
 			destination: recipientAddress,
 			asset: STELLAR.kinAsset,
-			amount
+			amount: amount.toString()
 		});
+		const memoText = `1-${this.appId}-${orderId}`;
```
Comment (nitzantomer) [inline scripts/src/mock_client.ts]: extract the version part to a const of something

File: scripts/src/utils.ts [hunk 6]
```diff
@@ -64,3 +64,11 @@ export function pick<T, K extends keyof T>(obj: T, ...props: K[]): Pick<T, K> {
 	props.forEach(name => newObj[name] = obj[name]);
 	return newObj;
 }
+
+export function removeDups(arr: string[]): string[] {
```
Comment (nitzantomer) [inline scripts/src/utils.ts]: better implementation:
```
export function removeDuplicates(arr: string[]): string[] {
    return Array.from(new Set(arr));
}
```

File: scripts/src/internal/app.ts (hunk 1 above)
Comment (doodyparizada) [reply scripts/src/internal/app.ts]: so it shows up in the log as msg=msg

File: scripts/src/internal/app.ts (hunk 2 above)
Comment (doodyparizada) [reply scripts/src/internal/app.ts]: same answer

**Pull request merged**